packages/api/lib/rulesHelpers.js

```diff
@@ -640,6 +641,9 @@ async function updateRuleTrigger(original, updates) {
     clonedRuleItem = await validateAndUpdateSqsRule(mergedRule);
     break;
   case 'onetime':
+    if (stateChanged && original.state === 'DISABLED') {
```

Comment:
I wouldn't want to run the rule upon changing the state.

I believe this would not only be confusing, but also represents a similar situation that this fix is attempting to avoid, which is execution at an inconvenient time. In other words, not only do I not want the rule to run when I initially create it with a DISABLED state, but I also don't want it to run when I simply change the state.

By immediately executing upon changing the state to ENABLED, the user cannot separately enable the rule in preparation for execution at a later time only. Further, this approach requires the user to know that all future executions of the enabled rule must now explicitly "rerun" the rule because setting the state to ENABLED will _not_ execute the rule again when its state is already ENABLED.

I feel that the implicit execution upon changing the state to ENABLED is problematic.

Simply change the state, and do not execute. Only "rerunning" the rule should actually run the rule.

Ideally, I would like to see no execution upon insertion of an ENABLED rule either, but I understand that making that change might represent a "breaking" change for existing users of "onetime" rules.

Reply:
Thanks for asking. Let me make sure I understand this part:

> For a onetime disabled rule, the only way to currently run the rule would be to run it through the invokeRerun(rerun endpoint)

Are you saying that a DISABLED rule can be executed by using the "rerun" endpoint?

If so, then I'm not sure that's a good idea. If a rule is DISABLED, then you should never be able to run it; neither automatically at creation time (which currently happens), nor at a later time. Only ENABLED rules should be allowed to run. Otherwise, the state value is a lie.

Therefore, in order to run a DISABLED rule, the user should first set its state to ENABLED (which should _not_ automatically run it), and then use the "rerun" endpoint to run it at a suitable time.